src/core/server/saved_objects/export/get_sorted_objects_for_export.ts

```diff
@@ -137,7 +140,7 @@ export async function getSortedObjectsForExport({
     exportSizeLimit,
     namespace,
   });
-  let exportedObjects = [...rootObjects];
+  let exportedObjects = [...rootObjects.sort(compareSavedObjects)];
```

Comment:
Since `exportedObjects` gets overriden in line 147 when `includeReferencesDeep = true` this won't sort when we include references.

The original code might be a bit easier to follow if we refactor it with an explicit else and perform the sort as the last step in the `sortObjects` function:
```ts
  let exportedObjects: SavedObject[] = [];
  let missingReferences: SavedObjectsExportResultDetails['missingReferences'] = [];
  if (includeReferencesDeep) {
    const fetchResult = await fetchNestedDependencies(rootObjects, savedObjectsClient, namespace);
    exportedObjects = sortObjects(fetchResult.objects);
    missingReferences = fetchResult.missingRefs;
  } else {
    exportedObjects = sortObjects(rootObjects);
  }
```